packages/astro/src/astro-components/control/ProtectCSR.astro

```diff
@@ -0,0 +1,20 @@
+---
+import type { ProtectProps } from '../../types';
+
+type Props = Omit<ProtectProps, 'condition'>;
+
+const { role, permission } = Astro.props;
+---
+
+<clerk-protect
+  data-role={role}
+  data-permission={permission}
+>
+  <!-- TODO: Can't have a data attribute in a Fragment. Remove div wrapper later. -->
```

Comment:
I guess you can remove that TODO? I checked Astro's docs a bit, seems like the div wrapper are necessary?

Reply:
Indeed. Removed the TODO and kept the `div`s 👍🏼